### DISCUSSION TITLE
Adapter lockup for CC2562/CC1352 base sticks fixed in latest dev release of Z-Stack

### DISCUSSION THREAD

**Discussion by walthowd** [Koenkk/zigbee2mqtt#6019] [Devices]
Just as a public service announcement, a long standing bug in Texas Instruments ZStack that caused  lockups and network shutdown on zigbee2mqtt looks like it has just been fixed. 

The bug would be very rare for some or most users, but more common for users with a large (100+) number of devices. The symptoms would be loss of communication with the zigbee coordinator radio. The serial port would still be present, but nothing would be returned from Z-Stack.

This bug affected:

*  ZZH 
* Slaesh's CC2562 stick
* Texas Instruments CC1352P2
* Texas Instruments CC26X2R1 
* Texas Instruments CC2538

Here is the link to the updated ZStack firmware (dated 1/25/21) that fixes the issue:

https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator/Z-Stack_3.x.0/bin

~~Just a note this is still in "dev" status and has not been mainlined yet.~~ 

Just merged into production 

Here are the Texas Instrument release notes:

http://software-dl.ti.com/simplelink/esd/simplelink_cc13x2_26x2_sdk/4.40.00.44/exports/changelog.html

> ZIGBEE-1172	NPI memory leak from buffer deallocation order

Update today and share your results if this issue has been affecting you.

**Comment by Koenkk** [comment]
Thanks! Wasnt aware of this, out of curiosity how do you know this change fixes this?

**Reply by walthowd** [reply]
Triggering it seems load based, in my personal setup it would trigger every 1 to 8 days. Working with the ZHA devs, setup a test platform where we could flood out ZDO.MgmtLQI requests and reliably trigger the lockup every 4 to 9 hours. 

Post ZStack upgrade that test platform has been running for over 36 hours now without any issues. 

And the behavior exactly mimics the original reported bug report by Egor here:
https://e2e.ti.com/support/wireless-connectivity/zigbee-and-thread/f/158/t/966975

and here:

https://e2e.ti.com/support/wireless-connectivity/zigbee-and-thread/f/158/t/973689

**Reply by puddly** [reply]
Same here. I was able to reliably reproduce this lockup bug by just continuously transmitting/receiving as fast as possible for 2-8 hours on the LAUNCHXL boards and with the ZZH. Did it about 4-5 times to be sure. After installing the firmware with this patch, I did not have any lockups for the ~40 hours I let the test run.

I believe there are a few open issues about Z2M just "not working after a while" that are likely caused by this bug. I wasn't able to see in the source code for Z-Stack Home 1.2 and Z-Stack 3.0.2 this same bug so they appear to be unaffected.

**Reply by choeflake** [reply]
Pairing a bunch of devices at the same time is also a good test scenario to trigger the bug. What a drama, partly paired devices, network id's of 0x00000000 etc.
